### PR TITLE
Sync e2e tests

### PR DESCRIPTION
**Pull request by cchudant**
# Pull Request type

- Testing

## Other information

End to end testing to sync a few blocks

### CODE REVIEW

**Comment by cchudant** [comment]
needs to be merged after #215 
I am trying to make github ci work also

File: .github/workflows/pull-request.yml [hunk 1]
```diff
@@ -30,7 +30,12 @@ jobs:
...
     needs: rust_check
 
-  rust_test:
-    name: Run Cargo tests
-    uses: ./.github/workflows/rust-test.yml
-    needs: rust_check
+  # rust_test:
+  #   name: Run Cargo tests
+  #   uses: ./.github/workflows/rust-test.yml
+  #   needs: rust_check
```
Comment (apoorvsadana) [inline .github/workflows/pull-request.yml]: let's delete if not needed

File: .github/workflows/push.yml [hunk 2]
```diff
@@ -20,7 +20,7 @@ jobs:
     uses: ./.github/workflows/linters-cargo.yml
     needs: rust_check
 
-  rust_test:
-    name: Run Cargo tests
-    uses: ./.github/workflows/rust-test.yml
-    needs: rust_check
```
Comment (apoorvsadana) [inline .github/workflows/push.yml:26]: same as above

File: crates/tests/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+            port: get_port(),
+        }
+    }
+
+    pub fn args(mut self, args: impl IntoIterator<Item = impl Into<String>>) -> Self {
+        self.args = args.into_iter().map(Into::into).collect();
+        self
+    }
+
+    pub fn env(mut self, args: impl IntoIterator<Item = impl Into<String>>) -> Self {
```
Comment (Mohiiit) [inline crates/tests/src/lib.rs]: the function is named env but it's modifying args

File: .github/workflows/coverage.yml
Comment (apoorvsadana) [inline .github/workflows/coverage.yml:1]: what's the difference between this and `rust-test.yml`, do we need both?

**Review by apoorvsadana — APPROVED** [review]

File: .github/workflows/coverage.yml
Comment (cchudant) [reply .github/workflows/coverage.yml:1]: probably not

File: Cargo.toml [hunk 4]
```diff
@@ -106,12 +108,8 @@ cairo-lang-starknet-classes = "=2.7.0-rc.2"
 cairo-lang-utils = "=2.7.0-rc.2"
 
 alloy = { version = "0.2.0", features = [
-  "node-bindings",
```
Comment (Mohiiit) [inline Cargo.toml:109]: node bindings is required to run the anvil

**Review by Mohiiit — APPROVED** [review]

File: .github/workflows/coverage.yml
Comment (cchudant) [reply .github/workflows/coverage.yml:1]: removed it

**Comment by cchudant** [comment]
:partying_face: :partying_face: :partying_face: ready to merge

**Review by antiyro — APPROVED** [review]

**Pull request merged**